### PR TITLE
Remove demo shit

### PR DESCRIPTION
**Pull request by KingTerrytheTerrible**
Reverts #12300 
fuck you steamport and especially FUCK YOU MONSTER860, what a laggy turd shitfest this is. waste of disk space

### CODE REVIEW

File: code/modules/admin/IsBanned.dm [hunk 1]
```diff
@@ -189,7 +189,7 @@
 			return null
 
 		if (C) //user is already connected!.
-			to_chat(C, "You are about to get disconnected for matching a sticky ban after you connected. If this turns out to be the ban evasion detection system going haywire, we will automatically detect this and revert the matches. if you feel that this is the case, please wait EXACTLY 6 seconds then reconnect using file -> reconnect to see if the match was automatically reversed.", confidential=TRUE)
+			to_chat(C, "You are about to get disconnected for matching a sticky ban after you connected. If this turns out to be the ban evasion detection system going haywire, we will automatically detect this and revert the matches. if you feel that this is the case, please wait EXACTLY 6 seconds then reconnect using file -> reconnect to see if the match was automatically reversed.")
```
Comment (Manly-Dwarf) [inline code/modules/admin/IsBanned.dm]: wat

Comment (KingTerrytheTerrible) [reply code/modules/admin/IsBanned.dm]: what do you mean what? it removes the demo subsystem

**Comment by AnthonyJNgo** [comment]
Decent concept but god this is terrible

**Comment by Toxici11i** [comment]
+1 literally no longer functions reliably, either needs to be disabled for the time being, or completely removed and reworked

**Comment by KingTerrytheTerrible** [comment]
steamport stop fucking marking shit as spam on MY FUCKING PR

**Comment by Toxici11i** [comment]
why was my comment marked as spam? The Replay system/logs have completely broken recently and have proved unreliable for some rounds, and otherwise do not save or load correctly.

The fact it was even marked as spam was turbogay. cease.
image

**Comment by mkikt4743** [comment]
yeah server is laggy all the time if this fixes it +1

**Comment by steamp0rt** [comment]
i marked toxic's +1 as spam because he's not a designer, maintainer, or admin, so his +1 doesn't really count for the PR.

**Comment by Toxici11i** [comment]
> i marked toxic's +1 as spam because he's not a designer, maintainer, or admin, so his +1 doesn't really count for the PR, AND I WANT TO GET THIS POINT THROUGH.

stop spam marking comments just because "they dont matter" when they arent actually spam, or arent trying to improve the discussion. You have marked both of my comments that contained some information that proved the extent of this terrible system.

To reiterate
--------
It doesnt work sometimes
It takes ages to work
It apparently eats lots of cpu
Its a terrible system that added lots of useless shit
It seemingly also causes other log files to fail to be uploaded correctly to statbus

just because my comment has a +1 in it doesnt make it invalid, faggot.

**Comment by steamp0rt** [comment]
@Toxici11i I'm fully aware of the flaws of the replay system. I'm also sure that you have no clue whatsoever how it works, and no clue why it may be causing this lag, other than some generic statement.

**Comment by Toxici11i** [comment]
> @Toxici11i I'm fully aware of the flaws of the replay system. I'm also sure that you have no clue whatsoever how it works, and no clue why it may be causing this lag, other than some generic statement.

what you mean the fact it tracks, logs and then has to write what every object does per tick, even objects that are doing nothing?

**Comment by mkikt4743** [comment]
bro you don't mark any other +1s from any sources on other PRs.... kinda cringe

**Comment by Manly-Dwarf** [comment]
lol i just find it funny to mark toxic's posts as spam

**Comment by Toxici11i** [comment]
> lol i just find it funny to mark toxic's posts as spam

faggot

**Comment by Toxici11i** [comment]
https://cdn.discordapp.com/attachments/276455180852264960/666024206018019359/Strike_Me_Down_Zeus.mp4 me ban baiting admins

**Comment by Pyko1** [comment]
-1 

It's currently disabled afaik, improve/fix instead of remove.
Or replace it with the rust based one that /tg/ is currently working on.

**Comment by KingTerrytheTerrible** [comment]
we cant "improve/fix" it, it's inherently flawed. yogs keeps OOMing because of it

**Comment by QuarianCommando** [comment]
+1 this shit is broken. Remove it now, replace when and if something better comes along.

**Comment by Pyko1** [comment]
I would still keep it for now, as there seem to be claims that its still lagging.
Even tho demos are disabled.

**Comment by QuarianCommando** [comment]
Why would we keep it if it's lagging?

**Comment by Pyko1** [comment]
if its still lagging, its suggesting its not the demos causing the lag, as they are disabled on server atm.

**Comment by karmaisblackandbluepilled** [comment]
ngl i've played a few rounds since it's been removed and the game is exactly as laggy so this didn't fix shit

**Comment by Pyko1** [comment]
I mean I haven't really seen any conclusive proof, or even reason why the demos are causing the lag.

So, I would keep the code in at least until we can conclude they were the are the source.
As it starting to seem like they aren't, as servers still lagging, and demos have been disabled for a week.

I mean we cant just start nuking random features since they might be cause of the lag, especially if the lag continues even after they are disabled.

Also AFAIK the lag and issues are not happening every round, and if the demos were the cause, then it would happen every round, as demos happen every round if they aren't disabled. Demos just add some overhead in top of things, but i recall we had lag issues already before we added demos.

**Comment by AnthonyJNgo** [comment]
I noticed considerable improvement after disabling the demo system in config, and I had around 20 other people online that also stated it was much better. The demo system was definitely part of the problem, however, I believe the root cause was the newest byond update. I've heard the new version of byond caused some issues on Goon initially as well until they changed a few things. It also caused issues with the portals pr that @steamp0rt made. Once Byond is updated and stable we'll be good, and we will **never** jump the gun on getting on to beta again.

**Comment by steamp0rt** [comment]
> It also caused issues with the portals pr that @steamp0rt made

My portal PR _only worked_ on 513 because it used new 513 features.

**Comment by AnthonyJNgo** [comment]
@steamp0rt I'm not saying your portals pr is the problem. I'm saying the root cause is 513 itself.

**Comment by KingTerrytheTerrible** [comment]
which we wouldnt have updated to if it wasnt for fucking steamport

**Comment by AnthonyJNgo** [comment]
That might be true, but there's nothing to gain from casting blame right now, all we can do is learn from our mistakes going forward. Now we know to never trust a beta build again haha.

**Comment by QuarianCommando** [comment]
@steamp0rt Even you admitted to me that while the portals PR wasn't the underlying cause of the crashes, it was a contributing factor.

**Comment by AnthonyJNgo** [comment]
Closing for now, demo is a config option and I'll play around with toggling it on/off now that 513 seems stable.

**Pull request closed without merging**